src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextWriter.cs

```diff
@@ -4,48 +4,67 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 using System.Linq;
+using System.Text.Encodings.Web;
+using System.Text.Json;
 
 namespace Microsoft.Extensions.DependencyModel
 {
-    public partial class DependencyContextWriter
+    public class DependencyContextWriter
     {
-        private void WriteCore(DependencyContext context, UnifiedJsonWriter jsonWriter)
+        public void Write(DependencyContext context, Stream stream)
         {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            // Custom encoder is required to fix https://github.com/dotnet/core-setup/issues/7137
+            // Since the JSON is only written to a file that is read by the SDK (and not transmitted over the wire),
+            // it is safe to skip escaping certain characters in this scenario
+            // (that would otherwise be escaped, by default, as part of defense-in-depth, such as +).
+            var options = new JsonWriterOptions { Indented = true, Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping };
+            var jsonWriter = new Utf8JsonWriter(stream, options);
```

Comment:
We need to dispose the writer. Put it in a using.
```suggestion
            using var jsonWriter = new Utf8JsonWriter(stream, options);
```